From 2d620c0206b716c2978d3920700080aa4af1c8cd Mon Sep 17 00:00:00 2001
From: Arulanbu Balusamy <quic_abalusam@quicinc.com>
Date: Thu, 15 Feb 2024 15:44:48 +0530
Subject: [PATCH] iw: Fix the null value updation on EHT MCS/NSS values.

The MCS/NSS values are not updated properly with the respective
bandwidth when the EHT-20 MHz is not applicable. So handle the
address increment of the mcs_set array properly and resolving
the Null value updation issue on the EHT bandwidths.

Signed-off-by: Arulanbu Balusamy <quic_abalusam@quicinc.com>
---
 util.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/util.c
+++ b/util.c
@@ -1624,6 +1624,7 @@ static void __print_eht_capa(int band,
 			printf("%s\t\tEHT bw=20 MHz, max NSS for MCS %s: Rx=%u, Tx=%u\n",
 			       pre, mcs[i],
 			       mcs_set[i] & 0xf, mcs_set[i] >> 4);
+		mcs_set += 4;
 	} else {
 		if (he_phy_cap[0] & (BIT(2) << 8)) {
 			for (i = 0; i < 3; i++)
